libs/isograph-react/src/loadable-hooks/useConnectionSpecPagination.ts

```diff
@@ -114,8 +114,13 @@ export function useConnectionSpecPagination<
         fragmentReference.readerWithRefetchQueries,
       );
 
+      const data = readOutDataAndRecords[i]?.item;
+      if (data == null) {
+        throw new Error('Parameter data is unexpectedly null');
```

Comment:
indicative of a bug in isograph, likewise below and in `useSkipLimit`. Could you also add a comment `// invariant: readOutDataAndRecords.length === completedReferences.length` somewhere near here as well?